### ISSUE TITLE
safe_int_mm's FakeTensor probe via input.__repr__() syncs the device on every eager int8 matmul and breaks CUDA graph capture

### ISSUE THREAD

**Issue by danielhanchen** [pytorch/ao#4872]
### Bug

`torchao/kernel/intmm.py::safe_int_mm` decides whether it is being traced with

```python
if dynamo_is_compiling() or "FakeTensor" in input.__repr__():
```

On a real CUDA tensor `Tensor.__repr__` formats the element values, which copies data to the host. Two consequences for the eager int8 path (`Int8DynamicActivationInt8WeightConfig`, and any checkpoint of those subclasses loaded with `torch.load`):

1. Every eager `int_scaled_matmul` call pays a device sync before the GEMM.
2. The linear cannot be captured with `torch.cuda.graph`: the `.item()` inside the repr raises `cudaErrorStreamCaptureUnsupported` and the capture is invalidated with `operation failed due to a previous error during capture`.

Present in 0.17.0 and 0.18.0 (`torchao/kernel/intmm.py` line 45 in 0.17.0).

### Repro

```python
import torch
from torchao.quantization import quantize_, Int8DynamicActivationInt8WeightConfig

lin = torch.nn.Linear(512, 512, dtype=torch.bfloat16, device="cuda")
quantize_(lin, Int8DynamicActivationInt8WeightConfig())
x = torch.randn(64, 512, dtype=torch.bfloat16, device="cuda")
with torch.inference_mode():
    s = torch.cuda.Stream(); s.wait_stream(torch.cuda.current_stream())
    with torch.cuda.stream(s):
        for _ in range(3): lin(x)
    torch.cuda.current_stream().wait_stream(s); torch.cuda.synchronize()
    g = torch.cuda.CUDAGraph()
    with torch.cuda.graph(g):   # raises: operation not permitted when stream is capturing
        lin(x)
```

Traceback ends in `torch/_tensor.py __format__ -> self.detach().item()` called from `torch/_tensor_str.py`, reached from `safe_int_mm` line 45.

### Suggested fix

Ask the question without touching the data:

```python
from torch._subclasses.fake_tensor import is_fake
if dynamo_is_compiling() or is_fake(input):
```

`is_fake` also unwraps `FunctionalTensor` and traceable wrapper subclasses, whose repr contains `FakeTensor` today, so it covers the same cases the string test did. With that one-line change the capture above succeeds and the graph replay is bit-identical to eager; I also verified the replacement bit-identical on the bare GEMM (aligned, misaligned K, non-contiguous `mat2`, CPU) and on a whole compiled diffusion transformer.

Environment: torchao 0.17.0+cu130, torch 2.12.1+cu130, B200 (sm_100), Python 3.13.
